Reconstruction/CRDEcalRec/src/Algorithm/GlobalClusteringAlg.cpp

```diff
@@ -52,6 +69,270 @@ StatusCode GlobalClusteringAlg::ClearAlgorithm(){
+{
+	std::vector<const PandoraPlus::CaloBar*> m_1dcluster = cluster1d->getCluster();
+	
+	for(int i1d = 0; i1d<m_1dcluster.size(); i1d++)
+	{
+		if(m_bar->getModule()==m_1dcluster[i1d]->getModule() && m_bar->getDlayer()==m_1dcluster[i1d]->getDlayer() && m_bar->getSlayer()==m_1dcluster[i1d]->getSlayer())
+		{
+			if(m_1dcluster[i1d]->getSlayer()==0)
+			{
+				if(m_1dcluster[i1d]->getBar()==1)
+				{
+					if(
+					(m_bar->getBar()==m_phibarnumber && m_bar->getStave()==(m_1dcluster[i1d]->getStave()-1) && abs(m_bar->getPart()-m_1dcluster[i1d]->getPart())<=1)||
+					((m_bar->getBar()==1 || m_bar->getBar()==2)  && m_bar->getStave()==m_1dcluster[i1d]->getStave() && abs(m_bar->getPart()-m_1dcluster[i1d]->getPart())<=1)
+					)
+					{
+						return true;
+					}
+				}
+				else if(m_1dcluster[i1d]->getBar()==m_phibarnumber)
+				{
+					if(
+					(m_bar->getBar()==1 && m_bar->getStave()==(m_1dcluster[i1d]->getStave()+1) && abs(m_bar->getPart()-m_1dcluster[i1d]->getPart())<=1)||
+					((m_bar->getBar()==m_phibarnumber || m_bar->getBar()==(m_phibarnumber-1))  && m_bar->getStave()==m_1dcluster[i1d]->getStave() && abs(m_bar->getPart()-m_1dcluster[i1d]->getPart())<=1)
+					)
+					{
+						return true;
+					}
+				}
+				else
+				{
+					if(abs(m_bar->getBar()-m_1dcluster[i1d]->getBar())<=1 && m_bar->getStave()==m_1dcluster[i1d]->getStave() && abs(m_bar->getPart()-m_1dcluster[i1d]->getPart())<=1)
+					{
+						return true;
+					}
+				}
+			}
+			else
+			{
+				if(m_1dcluster[i1d]->getBar()==1)
+				{
+					if(
+					(m_bar->getBar()==(m_zbarnumber-2*(m_bar->getDlayer()-1)) && m_bar->getPart()==(m_1dcluster[i1d]->getPart()-1) && abs(m_bar->getStave()-m_1dcluster[i1d]->getStave())<=1)||
+					((m_bar->getBar()==1 || m_bar->getBar()==2)  && m_bar->getPart()==m_1dcluster[i1d]->getPart() && abs(m_bar->getStave()-m_1dcluster[i1d]->getStave())<=1)
+					)
+					{
+						return true;
+					}
+				}
+				else if(m_1dcluster[i1d]->getBar()==(m_zbarnumber-2*(m_bar->getDlayer()-1)))
+				{
+					if(
+					(m_bar->getBar()==1 && m_bar->getPart()==(m_1dcluster[i1d]->getPart()+1) && abs(m_bar->getStave()-m_1dcluster[i1d]->getStave())<=1)||
+					((m_bar->getBar()==(m_zbarnumber-2*(m_bar->getDlayer()-1)) || m_bar->getBar()==((m_zbarnumber-2*(m_bar->getDlayer()-1))-1))  && m_bar->getPart()==m_1dcluster[i1d]->getPart() && abs(m_bar->getStave()-m_1dcluster[i1d]->getStave())<=1)
+					)
+					{
+						return true;
+					}
+				}
+				else
+				{
+					if(abs(m_bar->getBar()-m_1dcluster[i1d]->getBar())<=1 && m_bar->getPart()==m_1dcluster[i1d]->getPart() && abs(m_bar->getStave()-m_1dcluster[i1d]->getStave())<=1)
+					{
+						return true;
+					}
+				}
+			}
+		}		
+	}
+
+	
+	return false;
+}
+
+//1d->2d
+bool GlobalClusteringAlg::ifAdjacent(PandoraPlus::Calo1DCluster* m_1dcluster, PandoraPlus::Calo2DCluster* m_2dcluster)
+{
+	if(m_1dcluster->getBars().at(0)->getDlayer() == m_2dcluster->getCluster().at(0)->getCluster().at(0)->getDlayer())
```

Comment:
Need protection: if the bars/clusters vector is empty.

Reply:
Can be simplified as: 
`return (Dlayer1==Dlayer2 && ifSameTower(1Dclus, 2Dclus));`